### PR TITLE
Elixir travis

### PR DESCRIPTION
**Pull request by garrensmith**
## Overview

configure Travis to run the Elixir tests. Since all partition tests are in Elixir it's important we start getting them to run with each build. I've kept the elixir command separate from `make check` until we are able to run the Elixir tests on Jenkins.

## Testing recommendations

Travis should pass

## Checklist

- [ ] Code is written and works correctly;
- [ ] Changes are covered by tests;
- [ ] Documentation reflects the changes;

### CODE REVIEW

**Comment by wohali** [comment]
We have, apparently, a lot of failing elixir tests...they are more unstable than the old JavaScript ones because we are now trying to do race-y things on a 3-node cluster instead of a single node.

**Comment by iilyak** [comment]
> We have, apparently, a lot of failing elixir tests...they are more unstable than the old JavaScript ones because we are now trying to do race-y things on a 3-node cluster instead of a single node.

By looking at Jenkins log I see that there are some problems with elixir install.

**Comment by wohali** [comment]
@iilyak Jenkins isn't running Elixir tests yet - it's not in the default targets. Do you mean Travis?

File: .travis.yml [hunk 1]
```diff
@@ -2,10 +2,26 @@ language: erlang
...
-   - 21.2.3
-   - 20.3.8.5
-   - 19.3
+matrix:
+  include:
+  - language: elixir
+    elixir: 1.6
+    env: COMMAND=elixir
+    otp_release: 21.2.3
+  - language: erlang
```
Comment (wohali) [inline .travis.yml]: Eventually, we're going to want elixir tests running against all OTP releases. I see this as a temporary workaround, but it's ugly. Using the `before_script` to install elixir, or just installing elixir in all versions, might be a better approach. See other comments as well.

File: test/elixir/mix.exs [hunk 2]
```diff
@@ -5,7 +5,7 @@ defmodule Foo.Mixfile do
     [
       app: :foo,
       version: "0.1.0",
-      elixir: "~> 1.5",
+      elixir: "~> 1.6",
```
Comment (wohali) [inline test/elixir/mix.exs]: FYI Jenkins images right now have Elixir 1.6.6, so this should be fine.

**Pull request merged**